simulator/source/Combat_simulator.cpp

```diff
@@ -604,15 +604,14 @@ bool Combat_simulator::start_cast_slam(bool mh_swing, double rage, double& swing
     }
     if (use_sl)
     {
-        if (swing_time_left > config.combat.slam_spam_max_time)
+        if (mh_swing || swing_time_left > config.combat.slam_spam_max_time)
         {
             if ((mh_swing && rage > config.combat.slam_rage_dd) || rage > config.combat.slam_spam_rage)
             {
-                simulator_cout("Starting to channel slam.", " Latency: ", config.combat.slam_latency,"s");
-                slam_manager.queue_slam(time_keeper_.time + config.combat.slam_latency);
+                simulator_cout("Starting to cast slam.", " Latency: ", config.combat.slam_latency,"s");
+                slam_manager.cast_slam(time_keeper_.time + config.combat.slam_latency);
                 time_keeper_.global_cd = 1.5 + config.combat.slam_latency;
-                // We have started 'channeling' so set a value for the swing time for now which is larger than GCD
-                swing_time_left = 1.6 + config.combat.slam_latency;
+                swing_time_left = config.sim_time;
```

Comment:
if the slam cast never finishes (due to a bug), this should be imminently visible now